### DISCUSSION TITLE
Custom Layout with fixed header and footer : How to add a  plugin (a read-only plugin) that cannot be edited ?

### DISCUSSION THREAD

**Discussion by mikeljoe1** [react-page/react-page#1124] [Q&A]
I would like to understand how to add a cellPlugin which cannot be edited by the user. It should not have additional properties. It should be used/added as constants.

For example, the below control bar can be disabled so that users cannot modify anything.
![image](https://user-images.githubusercontent.com/56544314/148100067-0075d6f5-2432-411d-bc6b-eda900d62d2b.png)

Could you please let me know how the above control bar can be disabled for a plugin ?

Use case: Creating a custom layout with fixed header and footer which cannot be deleted/modified.

if you don't specify `controls`, no form is rendered in the bottom toolbar

**Comment by macrozone** [comment]
see the docs https://react-page.github.io/docs/#/custom-cell-plugins

here is an example:

```typescript
import type { CellPlugin } from '@react-page/editor';
import React from 'react';

const customLayoutPlugin: CellPlugin = {
  Renderer: ({ children }) => (
    <div>
     <Header />
      {children}
      <Footer />
    </div>
  ),
  id: 'my-custom-plugin',
  title: 'Custom plugin',
  description: 'Some custom plugin',
  version: 1,
};

export default customLayoutPlugin;



```

**Reply by mikeljoe1** [reply]
![image](https://user-images.githubusercontent.com/56544314/148107250-bf840ddd-d9d7-4f7c-a58b-acae2d1139e2.png)
A custom layout  can be created with a footer in place, but as I mentioned still the control bar loads for the same. This footer or custom layout cannot be edited. User can add only additional components to this layout.

**Reply by mikeljoe1** [reply]
![image](https://user-images.githubusercontent.com/56544314/148108186-eda62d41-2d73-41ad-9e03-921bc33b194e.png)

**Reply by mikeljoe1** [reply]
if you don't specify controls, no form is rendered in the bottom toolbar - Yes, I've not specified any controls but still the BottomToolbar pops up. Below is my code.


`import type { CellPlugin } from '@react-page/editor';
import React from 'react';
import FooterPlugin from "./FooterPlugin";


const customLayoutPlugin: CellPlugin<{

}> = {
  Renderer: ({ children }) => (
    <div
      style={{
        border: '1px solid black',
        backgroundColor: 'white',
        height: '100vh'
      }}
    >
      {children}
    </div>
  ),
  createInitialChildren: () => {
    return [
      [
        {
          plugin: FooterPlugin,
        }
      ],
    ];
  },
  id: 'page-layout-plugin',
  version: 1,
};
export default customLayoutPlugin;
`

**Reply by macrozone** [reply]
this part shows (useful) options for cell manipulation like clonging, deleting, hiding and moving the cell around. Why would you want to remove that?

**Reply by mikeljoe1** [reply]
Because, I'm trying to provide a fixed layout which cannot be modifed, what if the user change the position of the layout. i want to restrict how the content should be created.
Is there a way to disable/enable  options in the toolBar so that i can provide an option to delete the layout but it cannot be moved around.
